### PR TITLE
Seo for Messaging

### PR DESCRIPTION
**Pull request by zuobiao-zhou**
This PR do Seo for Messaging.

image
image
localhost_3000_docs_next_concepts-messaging_

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Comment by zuobiao-zhou** [comment]
cc @Anonymitaet

File: docs/concepts-messaging.md [hunk 1]
```diff
@@ -389,9 +395,9 @@ client.newProducer()
 
 ### Batching
 
-When batching is enabled, the producer accumulates and sends a batch of messages in a single request. The batch size is defined by the maximum number of messages and the maximum publish latency. Therefore, the backlog size represents the total number of batches instead of the total number of messages.
+In Pulsar, batching means the producer accumulates and sends a batch of messages in a single request. The batch size is defined by the maximum number of messages and the maximum publish latency. Therefore, the backlog size represents the total number of batches instead of the total number of messages.
 
-![Batching](/assets/batching.svg)
+![Batching mechanism in Pulsar](/assets/batching.svg)
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:400]:
```suggestion
![Batching in Pulsar](/assets/batching.svg)
```

File: docs/concepts-messaging.md [hunk 2]
```diff
@@ -389,9 +395,9 @@ client.newProducer()
 
 ### Batching
 
-When batching is enabled, the producer accumulates and sends a batch of messages in a single request. The batch size is defined by the maximum number of messages and the maximum publish latency. Therefore, the backlog size represents the total number of batches instead of the total number of messages.
+In Pulsar, batching means the producer accumulates and sends a batch of messages in a single request. The batch size is defined by the maximum number of messages and the maximum publish latency. Therefore, the backlog size represents the total number of batches instead of the total number of messages.
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:398]: Suggest keeping the original text unchanged since it's accurate

File: docs/concepts-messaging.md [hunk 3]
```diff
@@ -435,13 +441,13 @@ With message chunking enabled, when the size of a message exceeds the allowed ma
...
 
-![](/assets/chunking-01.png)
+![How Broker stores chunked message when there is only one producer](/assets/chunking-01.png)
 
 #### Handle interwoven chunked messages with one ordered consumer
 
 When multiple producers publish chunked messages into a single topic, the broker stores all the chunked messages coming from different producers in the same [managed ledger](concepts-architecture-overview.md#managed-ledgers). The chunked messages in the managed ledger can be interwoven with each other. As shown below, Producer 1 publishes message M1 in three chunks M1-C1, M1-C2 and M1-C3. Producer 2 publishes message M2 in three chunks M2-C1, M2-C2 and M2-C3. All chunked messages of the specific message are still in order but might not be consecutive in the managed ledger.
 
-![](/assets/chunking-02.png)
+![How Broker stores chunked message when there are mutiple producers](/assets/chunking-02.png)
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:450]:
```suggestion
![Interwoven chunked messages in Pulsar](/assets/chunking-02.png)
```

File: docs/concepts-messaging.md [hunk 4]
```diff
@@ -486,13 +492,20 @@ If no tenant or namespace is specified when a client creates a topic, the topic
...
+Pulsar namespaces are logical groupings of topics. A namespace is a logical nomenclature within a tenant. A tenant creates namespaces via the [admin API](admin-api-namespaces.md#create-namespaces). For instance, a tenant with different applications can create a separate namespace for each application. A namespace allows the application to create and manage a hierarchy of topics. The topic `my-tenant/app1` is a namespace for the application `app1` for `my-tenant`. You can create any number of [topics](#topics) under the namespace.
 
 ## Subscriptions
 
-A subscription is a named configuration rule that determines how messages are delivered to consumers. Four subscription types are available in Pulsar: [exclusive](#exclusive), [shared](#shared), [failover](#failover), and [key_shared](#key_shared). These types are illustrated in the figure below.
+A Pulsar subscription is a lease on a topic established by a group of consumers, it is a named configuration rule that determines how messages are delivered to consumers. There are four available subscription types in Pulsar:
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:499]: same as https://github.com/apache/pulsar-site/pull/671#discussion_r1287836845

Comment (Anonymitaet) [inline docs/concepts-messaging.md:499]:
```suggestion
A Pulsar subscription is a named configuration rule that determines how messages are delivered to consumers. It is a lease on a topic established by a group of consumers. There are four subscription types in Pulsar:
```

File: docs/concepts-messaging.md [hunk 5]
```diff
@@ -486,13 +492,20 @@ If no tenant or namespace is specified when a client creates a topic, the topic
...
+
+- [exclusive](#exclusive)
+- [shared](#shared)
+- [failover](#failover)
+- [key_shared](#key_shared) 
+
+These types are illustrated in the figure below.
 
-![Subscription types](/assets/pulsar-subscription-types.png)
+![Four Subscription types in Pulsar](/assets/pulsar-subscription-types.png)
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:508]:
```suggestion
![Subscription types in Pulsar](/assets/pulsar-subscription-types.png)
```

File: docs/concepts-messaging.md [hunk 6]
```diff
@@ -510,7 +523,7 @@ When a subscription has no consumers, its subscription type is undefined. The ty
 
 #### Exclusive
 
-In the *Exclusive* type, only a single consumer is allowed to attach to the subscription. If multiple consumers subscribe to a topic using the same subscription, an error occurs. Note that if the topic is partitioned, all partitions will be consumed by the single consumer allowed to be connected to the subscription.
+The Exclusive type is a subscription mode that only allows a single consumer to attach to the subscription. If multiple consumers subscribe to a topic using the same subscription, an error occurs. Note that if the topic is partitioned, all partitions will be consumed by the single consumer allowed to be connected to the subscription.
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:526]:
```suggestion
The exclusive type is a subscription type that only allows a single consumer to attach to the subscription. If multiple consumers subscribe to a topic using the same subscription, an error occurs. Note that if the topic is partitioned, all partitions will be consumed by the single consumer allowed to be connected to the subscription.
```

File: docs/concepts-messaging.md [hunk 7]
```diff
@@ -520,11 +533,11 @@ Exclusive is the default subscription type.
 
 :::
 
-![Exclusive subscriptions](/assets/pulsar-exclusive-subscriptions.svg)
+![Pulsar Exclusive subscriptions](/assets/pulsar-exclusive-subscriptions.svg)
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:536]:
```suggestion
![Exclusive subscription type in Pulsar](/assets/pulsar-exclusive-subscriptions.svg)
```

File: docs/concepts-messaging.md [hunk 8]
```diff
@@ -520,11 +533,11 @@ Exclusive is the default subscription type.
...
 
 #### Failover
 
-In the *Failover* type, multiple consumers can attach to the same subscription. 
+The Failover type is a subscription mode that multiple consumers can attach to the same subscription.
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:540]:
```suggestion
The failover type is a subscription type that multiple consumers can attach to the same subscription. 
```

File: docs/concepts-messaging.md [hunk 9]
```diff
@@ -560,7 +573,7 @@ A consumer is selected by running a module operation `mod (partition index, cons
       
       - for P1: Consumer D is the active consumer and Consumer C is the stand-by consumer.
 
-  ![Failover subscriptions](/assets/pulsar-failover-subscriptions-5.png)
+  ![Workflow of Pulsar Failover subscriptions when the number of partitions in a partitioned topic is less than the number of consumers](/assets/pulsar-failover-subscriptions-5.png)
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:576]:
```suggestion
  ![Workflow of failover subscription type in Pulsar](/assets/pulsar-failover-subscriptions-5.png)
```

File: docs/concepts-messaging.md [hunk 10]
```diff
@@ -572,7 +585,7 @@ A consumer is selected by running a module operation `mod (partition index, cons
   
   - P2, P5, and P8 are assigned to Consumer C. Consumer C is their active consumer. Consumer A and Consumer B are their stand-by consumers.
   
-  ![Failover subscriptions](/assets/pulsar-failover-subscriptions-1.svg)
+  ![Workflow of Pulsar Failover subscriptions when the number of partitions in a partitioned topic is greater than the number of consumers](/assets/pulsar-failover-subscriptions-1.svg)
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:588]:
```suggestion
  ![Workflow of failover subscription type in Pulsar](/assets/pulsar-failover-subscriptions-1.svg)
```
Keep it as concise as possible

File: docs/concepts-messaging.md [hunk 11]
```diff
@@ -584,7 +597,7 @@ A consumer is selected by running a module operation `mod (partition index, cons
   
   Consumer A is the master consumer, while consumer B would be the next consumer in line to receive messages if consumer A is disconnected.
 
-  ![Failover subscriptions](/assets/pulsar-failover-subscriptions-2.svg)
+  ![Workflow of Pulsar Failover subscriptions when there is one non-partitioned topic](/assets/pulsar-failover-subscriptions-2.svg)
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:600]:
```suggestion
  ![Workflow of failover subscription type in Pulsar](/assets/pulsar-failover-subscriptions-2.svg)
```

File: docs/concepts-messaging.md [hunk 12]
```diff
@@ -594,11 +607,11 @@ A consumer is selected by running a module operation `mod (partition index, cons
   
   - The non-partitioned topic 2 and non-partitioned topic 3 are assigned to consumer A. Consumer B is their stand-by consumer.
 
-  ![Failover subscriptions](/assets/pulsar-failover-subscriptions-3.svg)
+  ![Workflow of Pulsar Failover subscriptions when there are multiple non-partitioned topics](/assets/pulsar-failover-subscriptions-3.svg)
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:610]:
```suggestion
  ![Workflow of failover subscription type in Pulsar](/assets/pulsar-failover-subscriptions-3.svg)
```

File: docs/concepts-messaging.md [hunk 13]
```diff
@@ -594,11 +607,11 @@ A consumer is selected by running a module operation `mod (partition index, cons
...
 
 #### Shared
 
-In *shared* or *round robin* type, multiple consumers can attach to the same subscription. Messages are delivered in a round-robin distribution across consumers, and any given message is delivered to only one consumer. When a consumer disconnects, all the messages that were sent to it and not acknowledged will be rescheduled for sending to the remaining consumers.
+The Pulsar Shared subscriptions allow multiple consumers attach to the same subscription. Messages are delivered in a round-robin distribution across consumers, and any given message is delivered to only one consumer. When a consumer disconnects, all the messages that were sent to it and not acknowledged will be rescheduled for sending to the remaining consumers.
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:614]:
```suggestion
The shared subscription type in Pulsar allows multiple consumers to attach to the same subscription. Messages are delivered in a round-robin distribution across consumers, and any given message is delivered to only one consumer. When a consumer disconnects, all the messages that were sent to it and not acknowledged will be rescheduled for sending to the remaining consumers.
```

File: docs/concepts-messaging.md [hunk 14]
```diff
@@ -608,13 +621,13 @@ Shared subscriptions do not guarantee message ordering or support cumulative ack
 
 :::
 
-![Shared subscriptions](/assets/pulsar-shared-subscriptions.svg)
+![Pulsar Shared subscriptions](/assets/pulsar-shared-subscriptions.svg)
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:624]:
```suggestion
![Shared subscription type in Pulsar](/assets/pulsar-shared-subscriptions.svg)
```

File: docs/concepts-messaging.md [hunk 15]
```diff
@@ -608,13 +621,13 @@ Shared subscriptions do not guarantee message ordering or support cumulative ack
...
 
 #### Key_Shared
 
-In the *Key_Shared* type, multiple consumers can attach to the same subscription. Messages are delivered in distribution across consumers and messages with the same key or same ordering key are delivered to only one consumer. No matter how many times the message is re-delivered, it is delivered to the same consumer.
+The Pulsar Key_Shared subscriptions allow multiple consumers attach to the same subscription. But different with the Shared type, messages in Key_Shared type are delivered in distribution across consumers and messages with the same key or same ordering key are delivered to only one consumer. No matter how many times the message is re-delivered, it is delivered to the same consumer.
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:628]:
```suggestion
The Key_Shared subscription type in Pulsar allows multiple consumers to attach to the same subscription. But different with the Shared type, messages in the Key_Shared type are delivered in distribution across consumers and messages with the same key or same ordering key are delivered to only one consumer. No matter how many times the message is re-delivered, it is delivered to the same consumer.
```

File: docs/concepts-messaging.md [hunk 16]
```diff
@@ -608,13 +621,13 @@ Shared subscriptions do not guarantee message ordering or support cumulative ack
...
 
-![Key_Shared subscriptions](/assets/pulsar-key-shared-subscriptions.svg)
+![Pulsar Key_Shared subscriptions](/assets/pulsar-key-shared-subscriptions.svg)
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:630]:
```suggestion
![Key_Shared subscription type in Pulsar](/assets/pulsar-key-shared-subscriptions.svg)
```

File: docs/concepts-messaging.md [hunk 17]
```diff
@@ -941,13 +961,13 @@ For code examples, see [Java](client-libraries-java.md#multi-topic-subscriptions
...
 
-Normal topics are served only by a single broker, which limits the maximum throughput of the topic. *Partitioned topics* are a special type of topic handled by multiple brokers, thus allowing for higher throughput.
+Normal topics are served only by a single broker, which limits the maximum throughput of the topic. Partitioned topic is a special type of topic handled by multiple brokers, thus allowing for higher throughput.
 
 A partitioned topic is implemented as N internal topics, where N is the number of partitions. When publishing messages to a partitioned topic, each message is routed to one of several brokers. The distribution of partitions across brokers is handled automatically by Pulsar.
 
 The diagram below illustrates this:
 
-![](/assets/partitioning.png)
+![How Pulsar handles the distribution of partitions across brokers](/assets/partitioning.png)
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:970]:
```suggestion
![Partition distribution in Pulsar](/assets/partitioning.png)
```

File: docs/concepts-messaging.md [hunk 18]
```diff
@@ -1016,7 +1040,7 @@ Currently, non-persistent topics which are not partitioned are not persisted to
 
 ### Performance
 
-Non-persistent messaging is usually faster than persistent messaging because brokers don't persist messages and immediately send acks back to the producer as soon as that message is delivered to connected brokers. Producers thus see comparatively low publish latency with non-persistent topic.
+With persistent topics, all messages are durably persisted on disks, whereas with non-persistent topics, brokers don't persist messages and immediately send acks back to the producer as soon as that message is delivered to connected brokers, so Non-persistent messaging is usually faster than persistent messaging. Producers thus see comparatively low publish latency with non-persistent topic.
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:1043]:
```suggestion
With persistent topics, all messages are durably persisted on disks, whereas with non-persistent topics, brokers don't persist messages and immediately send acks back to the producer as soon as that message is delivered to connected brokers, so non-persistent messaging is usually faster than persistent messaging. Producers thus see comparatively low publish latency with non-persistent topics.
```

File: docs/concepts-messaging.md [hunk 19]
```diff
@@ -1084,7 +1108,7 @@ Apache Pulsar supports graceful failure handling and ensures critical data is no
 - Consumers get disconnected from the database or the HTTP server. When this happens, the database is temporarily offline while the consumer is writing the data to it and the external HTTP server that the consumer calls are momentarily unavailable.
 - Consumers get disconnected from a broker due to consumer crashes, broken connections, etc. As a consequence, unacknowledged messages are delivered to other available consumers.
 
-Apache Pulsar avoids these and other message delivery failures using at-least-once delivery semantics that ensure Pulsar processes a message more than once.
+Message redelivery in Apache Pulsar avoids these and other message delivery failures using at-least-once delivery semantics that ensure Pulsar processes a message more than once.
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:1111]: What does "these" refer to? Please specify them explicitly

File: docs/concepts-messaging.md [hunk 20]
```diff
@@ -1138,7 +1162,7 @@ In the second scenario at the bottom, the producer publishes message 1, which is
 
 ### Producer idempotency
 
-The other available approach to message deduplication is to ensure that each message is *only produced once*. This approach is typically called **producer idempotency**. The drawback of this approach is that it defers the work of message deduplication to the application. In Pulsar, this is handled at the [broker](reference-terminology.md#broker) level, so you do not need to modify your Pulsar client code. Instead, you only need to make administrative changes. For details, see [Managing message deduplication](cookbooks-deduplication.md).
+The other available approach is typically called **producer idempotency**, that is, each message is *only produced once* without data loss and duplicatuion. The drawback of this approach is that it defers the work of message deduplication to the application. In Pulsar, this is handled at the [broker](reference-terminology.md#broker) level, so you do not need to modify your Pulsar client code. Instead, you only need to make administrative changes. For details, see [Managing message deduplication](cookbooks-deduplication.md).
```
Comment (Anonymitaet) [inline docs/concepts-messaging.md:1165]:
```suggestion
The other available approach to message deduplication is **producer idempotency**, which means each message is *only produced once* without data loss and duplication. The drawback of this approach is that it defers the work of message deduplication to the application. In Pulsar, this is handled at the [broker](reference-terminology.md#broker) level, so you do not need to modify your Pulsar client code. Instead, you only need to make administrative changes. For details, see [Managing message deduplication](cookbooks-deduplication.md).
```

File: docs/concepts-overview.md [hunk 21]
```diff
@@ -2,6 +2,7 @@
 id: concepts-overview
 title: Pulsar Overview
 sidebar_label: "Overview"
+description: Apache Pulsar is a multi-tenant, high-performance solution for server-to-server messaging.
```
Comment (Anonymitaet) [inline docs/concepts-overview.md:5]:
```suggestion
description: Key features of Pulsar
```

**Comment by zuobiao-zhou** [comment]
fixed, PTAL. @Anonymitaet

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**